node/exts/erc20-bridge/signersvc/kwil.go

```diff
@@ -39,121 +46,336 @@ type EpochReward struct {
 	Amount    string
 }
 
-// erc20ExtAPI defines the ERC20 reward extension API used by SignerSvc.
-type erc20ExtAPI interface {
-	GetTarget() string
-	SetTarget(ns string)
-	InstanceInfo(tx context.Context) (*RewardInstanceInfo, error)
-	GetActiveEpochs(ctx context.Context) ([]*Epoch, error)
-	GetEpochRewards(ctx context.Context, epochID types.UUID) ([]*EpochReward, error)
-	VoteEpoch(ctx context.Context, epochID types.UUID, safeNonce int64, signature []byte) (string, error)
+// bridgeSignerClient defines the ERC20 bridge extension client for signer service.
+type bridgeSignerClient interface {
+	InstanceInfo(tx context.Context, namespace string) (*RewardInstanceInfo, error)
+	GetActiveEpochs(ctx context.Context, namespace string) ([]*Epoch, error)
+	GetEpochRewards(ctx context.Context, namespace string, epochID *types.UUID) ([]*EpochReward, error)
+	VoteEpoch(ctx context.Context, namespace string, txSigner auth.Signer, epochID *types.UUID, safeNonce int64, signature []byte) (types.Hash, error)
+}
+
+type txBcast interface {
+	BroadcastTx(ctx context.Context, tx *types.Transaction, sync uint8) (*types.ResultBroadcastTx, error)
+}
+
+type engineCall interface {
+	CallWithoutEngineCtx(ctx context.Context, db sql.DB, namespace, action string, args []any, resultFn func(*common.Row) error) (*common.CallResult, error)
+}
+
+type nodeApp interface {
+	AccountInfo(ctx context.Context, db sql.DB, account *types.AccountID, pending bool) (balance *big.Int, nonce int64, err error)
+	Price(ctx context.Context, dbTx sql.DB, tx *types.Transaction) (*big.Int, error)
 }
 
-type erc20rwExtApi struct {
-	clt       *client.Client
-	namespace string
+type DB interface {
+	sql.ReadTxMaker
+	sql.DelayedReadTxMaker
 }
 
-var _ erc20ExtAPI = (*erc20rwExtApi)(nil)
+type signerClient struct {
```

Comment:
Another way to implement signerClient is to call the `usersvc.Service`. This approach will have unnecessary wrap/unwrap the request/response types, should be easier to implement